src/SearchNamespace.js

```diff
@@ -0,0 +1,128 @@
+import GMusicNamespace from './GMusicNamespace';
+
+import Album from './Structs/Album';
+import Artist from './Structs/Artist';
+import Track from './Structs/Track';
+
+import { findContextPath } from './utils/context';
+
+export default class SearchNamespace extends GMusicNamespace {
+  constructor(...args) {
+    super(...args);
+    this.path = findContextPath();
+    if (this.path) {
+      this._watchForSearches();
+    }
+
+    this.addMethod('getSearchText', this.getSearchText.bind(this));
+    this.addMethod('getResults', this.getResults.bind(this));
+    this.addMethod('isSearching', this.isSearching.bind(this));
+    this.addMethod('playResult', this.playResult.bind(this));
+    this.addMethod('search', this.search.bind(this));
+  }
+
+  _text(elem, def) {
+    if (elem) {
+      return elem.textContent.trim();
+    }
+    return def;
+  }
+
+  _watchForSearches() {
+    const that = this;
+    let searchChanged = 0;
+
+    window.addEventListener('hashchange', () => {
+      if (!/^#\/sr\//g.test(window.location.hash)) return;
+      this.emitter.emit('change:search:text', this.getSearchText());
```

Comment:
`change:search:text` feels strange to me because it breaks the existing convention of throwing out namespaces for `change`. Maybe we can rename this to `change:search`?